lib/rules/no-potential-component-option-typo.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview detect if there is a potential typo in your component property
+ * @author IWANABETHATGUY
+ */
+'use strict'
+
+const utils = require('../utils')
+const vueComponentOptions = require('../utils/vue-component-options.json')
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow a potential typo in your component property',
+      categories: ['essential'],
+      recommended: false,
+      url: 'https://eslint.vuejs.org/rules/no-potential-component-option-typo.html'
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          presets: {
+            type: 'array',
+            items: {
+              type: 'string',
+              enum: ['all', 'vue', 'vue-router', 'nuxt']
+            },
+            uniqueItems: true,
+            minItems: 0
+          },
+          custom: {
+            type: 'array',
+            minItems: 0,
+            items: { type: 'string' },
+            uniqueItems: true
+          },
+          threshold: {
+            type: 'number',
+            'minimum': 1
+          }
+        }
+      }
+    ]
+  },
+
+  create: function (context) {
+    const option = context.options[0] || {}
+    const custom = option['custom'] || []
+    const presets = option['presets'] || []
+    const threshold = option['threshold'] || 1
+    let candidateOptions = []
+    if (presets.includes('all')) {
+      candidateOptions = Object.keys(vueComponentOptions).reduce((pre, cur) => {
+        return [...pre, ...vueComponentOptions[cur]]
+      }, [])
+    } else {
+      candidateOptions = presets.reduce((pre, cur) => {
+        return [...pre, ...vueComponentOptions[cur]]
+      }, [])
+    }
+    candidateOptions = [...new Set(candidateOptions.concat(custom))]
```

Comment:
Destructuring can be used.

```suggestion
    candidateOptions = [...new Set([...candidateOptions,...custom])]
```